### ISSUE TITLE
[BUG] Docstring formatting for all deep learners

### ISSUE THREAD

**Issue by TonyBagnall** [aeon-toolkit/aeon#579]
### Describe the bug

the docstrings for deep learning networks etc is not in the standard format so throws errors in docstring tests. Main issues is spacing and punctation, see below. Its one boring massive cut and paste job. I have done CNNClassifier in #572 so it passes but then went elsewhere. 

### Steps/Code to reproduce the bug

currently looks like this 
```python
    n_layers        : int, default = 3
        number of convolution layers
```
should look like this, single space either side of colon, capital letter for description and ending in a full stop.
```python
    n_layers : int, default = 2
        The number of convolution layers in the network.
```

### Expected results

pass doctests. Currently CNN is this on #572, still a few bugs but all reduction is good, the sheer number of warnings is I think occasionally crashing the test

WARNING: [numpydoc] Validation warnings while processing docstring for 'aeon.classification.deep_learning.CNNClassifier':
  ES01: No extended summary found
  PR03: Wrong parameters order. Actual: ('n_layers', 'kernel_size', 'n_filters', 'avg_pool_size', 'activation', 'padding', 'strides', 'dilation_rate', 'n_epochs', 'batch_size', 'callbacks', 'file_path', 'save_best_model', 'save_last_model', 'best_file_name', 'last_file_name', 'verbose', 'loss', 'metrics', 'random_state', 'use_bias', 'optimizer'). Documented: ('n_layers', 'kernel_size', 'n_filters', 'avg_pool_size', 'activation', 'padding', 'strides', 'dilation_rate', 'use_bias', 'random_state', 'n_epochs', 'batch_size', 'verbose', 'loss', 'optimizer', 'metrics', 'callbacks', 'file_path', 'save_best_model', 'save_last_model', 'best_file_name', 'last_file_name')
  PR06: Parameter "verbose" type should use "bool" instead of "boolean"
  PR06: Parameter "loss" type should use "str" instead of "string"
  PR07: Parameter "optimizer" has no description
  PR06: Parameter "metrics" type should use "str" instead of "string"
  PR07: Parameter "metrics" has no description
  SA01: See Also section not found

### Actual results

WARNING: [numpydoc] Validation warnings while processing docstring for 'aeon.classification.deep_learning.CNNClassifier':
  GL01: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
  ES01: No extended summary found
  PR01: Parameters {'optimizer', 'save_best_model', 'strides', 'n_epochs', 'padding', 'callbacks', 'loss', 'save_last_model', 'random_state', 'kernel_size', 'activation', 'best_file_name', 'file_path', 'avg_pool_size', 'verbose', 'dilation_rate', 'n_layers', 'metrics', 'batch_size', 'n_filters', 'last_file_name', 'use_bias'} not documented
  PR02: Unknown parameters {'save_last_model    ', 'best_file_name     ', 'kernel_size   ', 'metrics        ', 'verbose        ', 'n_epochs      ', 'file_path      ', 'n_filters      ', 'avg_pool_size  ', 'n_layers       ', 'activation     ', 'loss           ', 'save_best_model    ', 'padding        ', 'use_bias       ', 'optimizer      ', 'batch_size     ', 'last_file_name     ', 'dilation_rate  ', 'random_state   ', 'callbacks      ', 'strides        '}
  PR08: Parameter "n_layers       " description should start with a capital letter
  PR09: Parameter "n_layers       " description should finish with "."
  PR08: Parameter "kernel_size   " description should start with a capital letter
  PR09: Parameter "kernel_size   " description should finish with "."
  PR08: Parameter "n_filters      " description should start with a capital letter
  PR08: Parameter "avg_pool_size  " description should start with a capital letter
  PR09: Parameter "avg_pool_size  " description should finish with "."
  PR08: Parameter "activation     " description should start with a capital letter
  PR09: Parameter "activation     " description should finish with "."
  PR08: Parameter "padding        " description should start with a capital letter
  PR09: Parameter "padding        " description should finish with "."
  PR08: Parameter "strides        " description should start with a capital letter
  PR09: Parameter "strides        " description should finish with "."
  PR08: Parameter "dilation_rate  " description should start with a capital letter
  PR09: Parameter "dilation_rate  " description should finish with "."
  PR08: Parameter "use_bias       " description should start with a capital letter
  PR09: Parameter "use_bias       " description should finish with "."
  PR08: Parameter "random_state   " description should start with a capital letter
  PR09: Parameter "random_state   " description should finish with "."
  PR08: Parameter "n_epochs      " description should start with a capital letter
  PR09: Parameter "n_epochs      " description should finish with "."
  PR08: Parameter "batch_size     " description should start with a capital letter
  PR06: Parameter "verbose        " type should use "bool" instead of "boolean"
  PR08: Parameter "verbose        " description should start with a capital letter
  PR09: Parameter "verbose        " description should finish with "."
  PR06: Parameter "loss           " type should use "str" instead of "string"
  PR08: Parameter "loss           " description should start with a capital letter
  PR09: Parameter "loss           " description should finish with "."
  PR07: Parameter "optimizer      " has no description
  PR06: Parameter "metrics        " type should use "str" instead of "string"
  PR07: Parameter "metrics        " has no description
  PR08: Parameter "callbacks      " description should start with a capital letter
  PR09: Parameter "callbacks      " description should finish with "."
  PR07: Parameter "file_path      " has no description
  PR09: Parameter "save_best_model    " description should finish with "."
  PR09: Parameter "save_last_model    " description should finish with "."
  PR09: Parameter "best_file_name     " description should finish with "."
  PR09: Parameter "last_file_name     " description should finish with "."
  SA01: See Also section not found

**Comment by hadifawaz1999** [comment]
Will take care of this

**Comment by TonyBagnall** [comment]
partially fixed by #850 and #799 and not really a bug!
